### PR TITLE
rp2: init at 1.5.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/rp/rp2/package.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      inherit pname version;
+      sha256 = "sha256-ly7uqbDkurYN/FzctzeMx7peBwoLcoJ0bAGCxd4BH/E=";
+    };
+
+    propagatedBuildInputs = with python3Packages; [
+      wheel
+      lxml
+      nose
+    ];
+  };
```
Comment (FliegendeWurst) [inline pkgs/by-name/rp/rp2/package.nix:36]: Why not add these to python-modules?

File: pkgs/by-name/rp/rp2/package.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "eprbell";
+    repo = "rp2";
+    rev = version;
+    hash = "sha256-lNi2BBOZ5jX4NVufen0d+h2VZEwT51+B6nwM2xLPo7M=";
+  };
+
+  nativeBuildInputs = with python3Packages; [ setuptools ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/rp/rp2/package.nix:50]:
```suggestion
  build-system = with python3Packages; [ setuptools ];
```

File: pkgs/by-name/rp/rp2/package.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (FliegendeWurst) [inline pkgs/by-name/rp/rp2/package.nix:52]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/rp/rp2/package.nix [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    propagatedBuildInputs = with python3Packages; [
+      wheel
+      lxml
+      nose
+    ];
+  };
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "rp2";
+  version = "1.5.0";
```
Comment (FliegendeWurst) [inline pkgs/by-name/rp/rp2/package.nix:40]:
```suggestion
  version = "1.7.0";
```
released by now

**Comment by matthiasbeyer** [comment]
I am no longer interested in this, if someone wants to take over, feel free to take my commits.

**Pull request closed without merging**